code/datums/elements/decal.dm

```diff
@@ -5,10 +5,12 @@
 	var/description
 	var/mutable_appearance/pic
 
-/datum/element/decal/Attach(atom/target, _icon, _icon_state, _dir, _cleanable=FALSE, _color, _layer=TURF_LAYER, _description, _alpha=255)
+/datum/element/decal/Attach(atom/target, _icon, _icon_state, _dir, _cleanable=FALSE, _color, _layer=TURF_LAYER, _description, _alpha=255, mutable_appearance/_pic)
 	. = ..()
-	if(!isatom(target) || !generate_appearance(_icon, _icon_state, _dir, _layer, _color, _alpha, target))
+	if(!isatom(target) || ( !_pic && !generate_appearance(_icon, _icon_state, _dir, _layer, _color, _alpha, target)))
```

Comment:
I don't see the point of passing in the pic here, you have all information available already to reattach.

Reply:
like this?